### ISSUE TITLE
Callable defaults

### ISSUE THREAD

**Issue by Gr1N** [samuelcolvin/pydantic#155]
Hi,

`pydantic` works fine with defaults defined using annotations in case of values like:
```
class Test(BaseModel):
    k: uuid.UUID = uuid.uuid4()
```

But in many validation libraries possible to define callable defaults, with `pydantic` it's also possible but not works as expected from my point of view:
```
class Test(BaseModel):
    k: uuid.UUID = uuid.uuid4

test = Test()
assert isinstance(test.k, uuid.UUID)
```
This code will fail because `test.k` is `<bound method uuid4 of <Test >>`.

Is it possible to fix this issue?

**Comment by samuelcolvin** [comment]
yes, use a validator with `alwasy=True` and return `uuid.uuid4()`

**Comment by Gr1N** [comment]
Read docs first :) Thanks

**Comment by samuelcolvin** [comment]
No problem, is not that obvious in the docs
